client/src/components/EventItem.js

```diff
@@ -80,76 +71,82 @@ const showToastMessageSuccesfull = (message) =>
-              },
-              body: JSON.stringify(editedEvent),
-              mode: "cors"
-          })
-              .then(response =>{
-                  if(response.status === 409){
-                    console.log("tickets should be more than the tickets already sold");
-                    showToastMessage("tickets should be more than the tickets already sold");
-                  }
-                  return response.json()
-              } )
-              .then(data => {
-                if (data) { 
-                  showToastMessageSuccesfull(`${data.event.ticketsSold} users will be notified of the changes`);
-                }
-              })
-
-          // Empty the input fields
-          setEditedEvent({});
-          // Close the Modal
-          setEdit(false);
-          if(checkedTicket) {
-            resetTickets();
-          }
-          if(checkedDeadline) {
-            handleDeadlineSwitch();
-          }
-          } else {
-            console.log("Starting time needs to be before ending time.");
-            showToastMessage("Starting time needs to be before ending time.");
-          }
-        } else {
-          console.log("Event joining deadline needs to be before or the same as the starting time of the event.");
-            showToastMessage("Event joining deadline needs to be before or the same as the starting time of the event.");
+    e.preventDefault();
+    let errorMessage = savingRules();
+    let successToast = false; //Doesn't work currently.
+    if (errorMessage != ""){
+      console.log(errorMessage);
+      showToastMessage(errorMessage);
+      return;
+    }
+    console.log(checkedDeadline);
+    if(!checkedDeadline) {
+      editedEvent.joinDeadline = undefined;
+    }
+    editedEvent.creator = props.currentUser.firstname + " " + props.currentUser.lastname;
+    editedEvent.creatorId = props.currentUser.id;
+    editedEvent.attendees = 1;
+    editedEvent.id = props.eventID;
+    
+    setEditedEvent(editedEvent);
+    fetch("/api/editEvent", { // TODO change url to event edit url
+      method: "POST",
+      headers: {
+          "Content-type": "application/json",
+          "Authorization": "Bearer " + props.currentUser.token
+      },
+      body: JSON.stringify(editedEvent),
+      mode: "cors"
+    })  
+    .then(response =>{
+        if(response.status === 409){
+          errorMessage = "tickets should be more than the tickets already sold";
+          console.log(errorMessage);
+          showToastMessage(errorMessage);
+          return;
         }
-      } else {
-        console.log("Ending time is not valid.");
-        showToastMessage("Ending time is not valid.");
+        return response.json();
+    } )
+    .then(data => {
+      if (data) { 
+        // if(! successToast){
+        //   showToastMessageSuccesfull(`${data.event.ticketsSold} users will be notified of the changes`);
+        //   successToast = true;
+        // }
```

Comment:
Here, the toast is displayed twice for some reason, and it produces an error when they disappear (something like "v.get() is undefined"). So I commented it for the moment because I think that's a tricky bug that's going to take a lot of time to solve and that doesn't bring that much value to the project